autoload/codi.vim

```diff
@@ -879,8 +891,44 @@ function _G.codi_select(interpreters)
     vim.fn["codi#new"](ft)
   end)
 end
+
+function _G.codi_expand_popup(lines)
+  local col = vim.g["codi#virtual_text_pos"]
+  local posx
+  if type(col) == "number" then
+    posx = col + #vim.g["codi#virtual_text_prefix"]
+  elseif col == "right_align" then
+    posx = vim.fn.winwidth(0)
+  else
+    posx = #vim.fn.getline(".") + #vim.g["codi#virtual_text_prefix"]
+  end
+
+  vim.lsp.util.open_floating_preview(lines, "", {
+    wrap = false,
+    border = "rounded",
+    offset_x = posx - vim.fn.wincol() + 1,
+  })
+end
 EOF
 
 function! codi#select()
   call v:lua.codi_select(s:interpreters)
 endfunction
+
+function! codi#expand()
+  let lineidx = line(".") - 1
+  if lineidx >= len(s:results)
+    return
+  endif
+  let lines = s:results[lineidx]
+  if len(lines) ==  0
+    return
+  endif
+
+  if has("nvim")
```

Comment:
Since we do the `has("nvim")` check in `plugin/codi.vim`, can we remove it from here?

Reply:
We could, but I kept it here to help someone who could want to implement a vim version. I can remove it if you want.

Reply:
I see, keeping it here sounds fine!